debezium-core/src/main/java/io/debezium/converters/spi/RecordParser.java

```diff
@@ -51,8 +52,21 @@ protected RecordParser(Schema schema, Struct record, String... dataFields) {
         this.dataSchema = getDataSchema(schema, connectorType, dataFields);
     }
 
+    protected RecordParser(Schema schema, Struct record, Struct source, SchemaAndValue operation, Struct transaction) {
```

Comment:
Instead of adding a new constructor I'd propose to change the existing one to with one parameter of type interface like
```
interface RecordAndMetadata {
  Struct getRecord();
  Struct source();
  String operation();
  Struct transaction();
}
```
there will be two implementations for this interface, one operating over `source` info block and the other over headers.
WDYT?

Reply:
done